airbyte-integrations/connectors/source-notion/source_notion/source.py

```diff
@@ -33,10 +34,21 @@ def get_access_token(self):
 
 
 class SourceNotion(AbstractSource):
-    def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> Tuple[bool, any]:
+    @staticmethod
+    def set_start_date(config: Mapping[str, Any]):
+        """
+        Sets the start date to two years ago if it is not already set by the user.
+        """
+        start_date = config.get("start_date")
+        if not start_date:
+            two_years_ago = pendulum.now().subtract(years=2).in_timezone("UTC").format("YYYY-MM-DDTHH:mm:ss.SSS[Z]")
+            config["start_date"] = two_years_ago
+
```

Comment:
replace extra class `NotionAuthenticator` with method:  
```suggestion
def _get_authenticator(self, config: Mapping[str, Any]):
    if "access_token" in self.config:
        token = self.config.get("access_token")
    elif:  # TODO: add else cases
        ...
    return TokenAuthenticator(token)
```

Reply:
Hi Artem, thanks for the feedback, I've made the changes to the authentication handling as requested.

Reply:
LGTM, please check `# support the old config`.
Will approve after that.